#### Sensor's name #####
imu_sensor_name: artus_LSM6DSOX_LISMDL

#### USB port #####
serial_port: /dev/ttyACM3

##### Calibration ####
## ~~ TODO: Currently Hardcoded in Arduino Firmwareâ€¦ ~~ ##
### --- accelerometer --- ###
accOffset_x:  0.00080732
accOffset_y:  0.012271
accOffset_z:  0.0052966

accGain_00:  0.99455
accGain_01:  0.0024532
accGain_02:  0.0030739

accGain_10: -0.013746
accGain_11:  1.0016
accGain_12: -0.022257

accGain_20:  0.029301
accGain_21:  0.046798
accGain_22:  0.99584

### --- Gyroscope --- ###
gyroOffset_x: -0.36530
gyroOffset_y:  0.16497
gyroOffset_z: -0.58260

### --- Magnetometer --- ###
magOffset_x: -80.78
magOffset_y:  26.91
magOffset_z: -46.47

magGain_00:  0.98
magGain_01:  0.044
magGain_02:  -0.003

magGain_10:  0.044
magGain_11:  1.045
magGain_12:  0.013

magGain_20: -0.003
magGain_21:  0.013
magGain_22:  0.979

##### Covariance Matrix ####
#### ~> !!! currently DummyValues !!!
# Orientation covariance estimation:
# Observed orientation noise: 0.3 degrees in x, y, 0.6 degrees in z
# Magnetometer linearity: 0.1% of full scale (+/- 2 gauss) => 4 milligauss
# Earth's magnetic field strength is ~0.5 gauss, so magnetometer nonlinearity could
# cause ~0.8% yaw error (4mgauss/0.5 gauss = 0.008) => 2.8 degrees, or 0.050 radians
# i.e. variance in yaw: 0.0025
# Accelerometer non-linearity: 0.2% of 4G => 0.008G. This could cause
# static roll/pitch error of 0.8%, owing to gravity orientation sensing
# error => 2.8 degrees, or 0.05 radians. i.e. variance in roll/pitch: 0.0025
# so set all covariances the same.
orientation_cov_00:  -1.0
#orientation_cov_00:  0.0025
orientation_cov_01:  0.0
orientation_cov_02:  0.0

orientation_cov_10:  0.0
orientation_cov_11:  0.0025
orientation_cov_12:  0.0

orientation_cov_20:  0.0
orientation_cov_21:  0.0
orientation_cov_22:  0.0025


# Angular velocity covariance estimation:
# Observed gyro noise: 4 counts => 0.28 degrees/sec
# nonlinearity spec: 0.2% of full scale => 8 degrees/sec = 0.14 rad/sec
# Choosing the larger (0.14) as std dev, variance = 0.14^2 ~= 0.02
angularVelocity_cov_00:  -1.0
#angularVelocity_cov_00:  0.02
angularVelocity_cov_01:  0.0
angularVelocity_cov_02:  0.0

angularVelocity_cov_10:  0.0
angularVelocity_cov_11:  0.02
angularVelocity_cov_12:  0.0

angularVelocity_cov_20:  0.0
angularVelocity_cov_21:  0.0
angularVelocity_cov_22:  0.02

# linear acceleration covariance estimation:
# observed acceleration noise: 5 counts => 20milli-G's ~= 0.2m/s^2
# nonliniarity spec: 0.5% of full scale => 0.2m/s^2
# Choosing 0.2 as std dev, variance = 0.2^2 = 0.04
linearAcc_cov_00:  -1.0
#linearAcc_cov_00:  0.04
linearAcc_cov_01:  0.0
linearAcc_cov_02:  0.0

linearAcc_cov_10:  0.0
linearAcc_cov_11:  0.04
linearAcc_cov_12:  0.0

linearAcc_cov_20:  0.0
linearAcc_cov_21:  0.0
linearAcc_cov_22:  0.04